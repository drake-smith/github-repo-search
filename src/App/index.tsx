import React from 'react';
import styled from 'styled-components';
import { media } from '../shared/media';
import SearchBar from '../SearchBar';

const { tablet, desktop } = media;

const PageContainer = styled.div`
  height: 100vh;
  width: 100%;
  padding: 1.5rem;

  ${tablet`
    padding: 2rem;
  `}

  ${desktop`
    padding: 3rem;
  `};
`;

const ContentContainer = styled.div`
  max-width: 1280px;
  margin: 0 auto;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`;

const Title = styled.h1`
  font-size: 1.5rem;
  font-weight: bold;
  font-style: italic;
  background-color: #ed1c24;
  color: #ffffff;
  display: inline-block;
  padding: 0.5rem 1rem;

  ${tablet`
    font-size: 3rem;
    padding: 0.5rem 1.5rem;
  `}

  ${desktop`
    font-size: 4rem;
  `};
`;

const App = () => {
  return (
    <PageContainer>
      <ContentContainer>
        <Title>GitHub Repo Finder</Title>
        <SearchBar
          handleSearch={() => {
            console.log('Ran!');
            return;
          }}
          isDisabled={false}
        />
      </ContentContainer>
    </PageContainer>
  );
};

export default App;
