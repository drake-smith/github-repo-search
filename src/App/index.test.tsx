import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import App from '.';

// Mock fetch object
global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () =>
      Promise.resolve({
        items: [
          {
            id: 1,
            name: 'React',
          },
          {
            id: 2,
            name: 'Angular',
          },
        ],
      }),
  })
) as jest.Mock;

describe('<App />', () => {
  test('<App /> renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <App />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
