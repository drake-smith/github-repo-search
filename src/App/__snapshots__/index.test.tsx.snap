// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<div
  className="sc-hKwDye dnRGEm"
>
  <div
    className="sc-eCImPb cCCSIQ"
  >
    <h1
      className="sc-jRQBWg eywaeu"
    >
      GitHub Repo Finder
    </h1>
    <form
      className="sc-bdvvtL ilklID"
      onSubmit={[Function]}
    >
      <input
        className="sc-gsDKAQ ifTxWv"
        onChange={[Function]}
        placeholder="Search for GitHub repositories"
        type="search"
        value=""
      />
      <input
        className="sc-dkPtRN eshZZB"
        disabled={false}
        type="submit"
        value="Search"
      />
    </form>
  </div>
</div>
`;
